tests/unit_tests/blas/batch/omatadd_batch_stride.cpp

```diff
@@ -0,0 +1,232 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#include <algorithm>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include <limits>
+#include <vector>
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+#include "allocator_helper.hpp"
+#include "cblas.h"
+#include "oneapi/mkl/detail/config.hpp"
+#include "oneapi/mkl.hpp"
+#include "onemkl_blas_helper.hpp"
+#include "reference_blas_templates.hpp"
+#include "test_common.hpp"
+#include "test_helper.hpp"
+
+#include <gtest/gtest.h>
+
+using namespace sycl;
+using std::vector;
+
+extern std::vector<sycl::device *> devices;
+
+namespace {
+
+template <typename fp>
+int test(device *dev, oneapi::mkl::layout layout, int64_t batch_size) {
+    // Prepare data.
+    int64_t m, n;
+    int64_t lda, ldb, ldc;
+    oneapi::mkl::transpose transa, transb;
+    fp alpha, beta;
+    int64_t i, tmp;
+
+    batch_size = 1 + std::rand() % 20;
```

Comment:
I am unsure this be the same random number every time.    

Do we want reproducibility of random batch size?

Reply:
In general, if something in a test were to fail.   Is knowing the exact values useful for reproducing.

Reply:
In this case, transpose APIs, there is probably not a concern.